src/home_robot/home_robot/agent/motion/stretch.py

```diff
@@ -0,0 +1,670 @@
+MANIP_STRETCH_URDF = "assets/hab_stretch/urdf/planner_calibrated_manipulation_mode.urdf"
+
+STRETCH_HOME_Q = np.array(
+    [
+        0,  # x
+        0,  # y
+        0,  # theta
+        0.2,  # lift
+        0.057,  # arm
+        0.0,  # gripper rpy
+        0.0,
+        0.0,
+        3.0,  # wrist,
+        0.0,
+        0.0,
+    ]
+)
+STRETCH_PREGRASP_Q = np.array(
+    [
+        0,  # x
+        0,  # y
+        0,  # theta
+        0.6,  # lift
+        0.01,  # arm
+        0.0,  # gripper rpy
+        3.14,  # wrist roll
+        -1.57,  # wrist pitch
+        0.0,  # wrist yaw
+        0.0,
+        0.0,
+    ]
+)
+
+# This is the gripper, and the distance in the gripper frame to where the fingers will roughly meet
+STRETCH_GRASP_FRAME = "link_straight_gripper"
+STRETCH_STANDOFF_DISTANCE = 0.235
+STRETCH_STANDOFF_WITH_MARGIN = 0.25
+# Offset from a predicted grasp point to STRETCH_GRASP_FRAME
+STRETCH_GRASP_OFFSET = np.eye(4)
+STRETCH_GRASP_OFFSET[:3, 3] = np.array([0, 0, -1 * STRETCH_STANDOFF_DISTANCE])
+# Offset from STRETCH_GRASP_FRAME to predicted grasp point
+STRETCH_TO_GRASP = np.eye(4)
+STRETCH_TO_GRASP[:3, 3] = np.array([0, 0, STRETCH_STANDOFF_DISTANCE])
+STRETCH_GRIPPER_OPEN = 0.22
+STRETCH_GRIPPER_CLOSE = -0.2
+# STRETCH_GRIPPER_CLOSE = -0.5
+
+
+class HelloStretchIdx:
+    BASE_X = 0
+    BASE_Y = 1
+    BASE_THETA = 2
+    LIFT = 3
+    ARM = 4
+    GRIPPER = 5
+    WRIST_ROLL = 6
+    WRIST_PITCH = 7
+    WRIST_YAW = 8
+    HEAD_PAN = 9
+    HEAD_TILT = 10
+
+
+class HelloStretch(Robot):
+    """define motion planning structure for the robot"""
+
+    # DEFAULT_BASE_HEIGHT = 0.09
+    DEFAULT_BASE_HEIGHT = 0
+    GRIPPER_OPEN = 0.6
+    GRIPPER_CLOSED = -0.3
+
+    default_step = np.array(
+        [
+            0.1,
+            0.1,
+            0.2,  # x y theta
+            0.025,
+            0.025,  # lift and arm
+            0.3,  # gripper
+            0.1,
+            0.1,
+            0.1,  # wrist rpy
+            0.2,
+            0.2,  # head
+        ]
+    )
+    default_tols = np.array(
+        [
+            0.1,
+            0.1,
+            0.01,  # x y theta
+            0.001,
+            0.0025,  # lift and arm
+            0.01,  # gripper
+            0.01,
+            0.01,
+            0.01,  # wrist rpy
+            10.0,
+            10.0,  # head - TODO handle this better
+        ]
+    )
+    # look_at_ee = np.array([-np.pi/2, -np.pi/8])
+    look_at_ee = np.array([-np.pi / 2, -np.pi / 4])
+    look_front = np.array([0.0, -np.pi / 4])
+    look_ahead = np.array([0.0, 0.0])
+
+    # For inverse kinematics mode
+    ee_link_name = "link_straight_gripper"
+    manip_mode_controlled_joints = [0, 3, 4, 5, 6, 7, 8, 9, 10]
+    # manip_mode_controlled_joints = [0, 5, 6, 7, 8, 9, 10, 11, 12]
+    # manip_mode_controlled_joints = [0, 7, 8, 9, 10, 11, 12, 13, 14]
+    full_body_controlled_joints = [0, 1, 2, 5, 6, 7, 8, 9, 10, 11, 12]
```

Comment:
We should move away from using joint indices to specify controllable joints because every time we create an IK object with a new control mode we need to go through the tedious process of mapping joints to their indices again.

Can `controlled_joints` be a list of joint names, which is easier to specify by just going through the urdf, and also gives more visibility into what's happening?
We can parse the list of joint names into indices within `PybulletIKSolver` by"
```py
joint_name_to_idx_map = {pb.getJointInfo(j, ...)[1]: j for j in range(pb.getNumJoints(...))}
controlled_joints_idcs = [joint_name_to_idx_map[joint_name] for joint_name in controlled joints]
```

Reply:
done - incorporated into the bullet API